contracts/modules/v2/Assessment.sol

```diff
@@ -0,0 +1,708 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-v4/utils/cryptography/MerkleProof.sol";
+import "../../interfaces/INXMToken.sol";
+import "../../abstract/MasterAwareV2.sol";
+import "hardhat/console.sol";
+
+/**
+ *  Provides a way for cover owners to submit claims and redeem the payouts and facilitates
+ *  assessment processes where members decide the outcome of the events that lead to potential
+ *  payouts.
+ */
+contract Assessment is MasterAwareV2 {
+
+  /* ========== DATA STRUCTURES ========== */
+
+  enum PollStatus { PENDING, ACCEPTED, DENIED }
+
+  enum EventType { CLAIM, INCIDENT }
+
+  enum Asset { ETH, DAI }
+
+  enum UintParams {
+    REWARD_PERC,
+    FLAT_ETH_FEE_PERC,
+    INCIDENT_TOKEN_WEIGHT_PERC,
+    VOTING_PERIOD_DAYS_MIN,
+    VOTING_PERIOD_DAYS_MAX,
+    PAYOUT_COOLDOWN_DAYS
+  }
+
+  struct Stake {
+    uint104 amount;
+    uint104 voteRewardCursor;
+    uint16 fraudCount;
+    /*uint32 unused,*/
+  }
+
+  /**
+   *  Holds data for a vote belonging to an assessor.
+   *
+   *  @dev This structure is used to keep track of user's votes. Each vote is used to determine
+   *  a user's share of rewards or to create a fraud resolution which excludes fraudulent votes
+   * from the initial poll.
+   *
+   *  @param eventId      Can be either a claimId or an IncidentId
+   *  @param accepted     If the assessor voted to accept the event it's true otherwise it's false
+   *  @param timestamp    Date and time when the vote was cast
+   *  @param tokenWeight  How many tokens were staked when the vote was cast
+   *  @param eventType    Can be a claim or an incident (See EventType enum)
+   */
+  struct Vote {
+    uint104 eventId;
+    bool accepted;
+    uint32 timestamp;
+    uint104 tokenWeight;
```

Comment:
this isn't a "weight", but rather the number of tokens that were voted with; it confused me a bit.
I'd call it `stake`

Reply:
https://en.wikipedia.org/wiki/Weighted_voting

Votes are not created equal because a vote from someone with a stake of 100 NXM **weighs** more than a vote from someone with 10 NXM staked. The term weight was also used in the proposal doc and explained in the annotation above. Even if tokenWeight is nothing more than a snapshot of the user's stake, in the context of votes it makes more sense to call it weight as it better describes the purpose of the value (determining how important a vote is rather than what the stake of a voter was) and it also dispels any potential confusion when it's used in formulas that also involve stake in their terms.